s/api/services/v1/comment-writing-service.ts

```diff
@@ -102,41 +102,44 @@ export const makeCommentWritingService = asServiceProvider(({
 		})
 	},
 
-	async archiveComment(rawId: string): Promise<void> {
-		const id = enforceValidation(rawId, validateId)
+	async archiveComments(rawIds: string[]): Promise<void> {
+		const idArr = enforceValidation(rawIds, validateIdArray)
 
 		if (!user)
 			throw new renraku.ApiError(403, "cannot archive, not logged in")
 
-		const specificComment = await database.tables.comments.readOne(
-			find({id: dbmage.Id.fromString(id)})
-		)
-
-		if (!specificComment)
-			throw new renraku.ApiError(404, "cannot archive, comment not found")
-
-		const userIsTheAuthor = user.id === specificComment.authorId.string
-		const userHasAdminRights = user.permissions.canArchiveAnyComment
-		const userIsAllowedToArchive = userIsTheAuthor || userHasAdminRights
-
-		if (!userIsAllowedToArchive) 
-			throw new renraku.ApiError(403, "forbidden, must be author or admin to archive a comment")
+		const ids = idArr.map(id => dbmage.Id.fromString(id))
+		for (const id of idArr) {
+			const specificComment = await database.tables.comments.readOne(
```

Comment:
this line will have unacceptably bad performance

here we're looping over an array of ids (there could be 100 or more!), and for each of those, awaiting an independent database query (readOne).

instead, we should use `findAll` to launch a single `read` query for all of the ids, to fetch all those comments.

Reply:
okay this is noted.
Thanks a lot @chase-moskal